src/components/modal/modal.tsx

```diff
@@ -0,0 +1,116 @@
+import { FC, Fragment, useRef } from 'react';
+import { ContentVariant, ModalProps, WidthModal } from './types';
+import { Dialog, Transition } from '@headlessui/react';
+import { Variant } from '../../utlis';
+import { Button, ButtonSize } from '../button';
+import { IconCancel, IconCheckmark, IconSize } from '../../elements';
+import { Heading, HeadingSize } from '../typography';
+import clsx from 'clsx';
+
+export const Modal: FC<ModalProps> = ({
+    isOpen = false,
+    title,
+    children,
+    size = WidthModal.MEDIUM,
+    contentVariant = ContentVariant.SETTINGS,
+    onActionPrimary,
+    onActionSecondary,
+    buttonLabelPrimary,
+    buttonLabelSecondary,
+}) => {
+    const initialFocusInputRef = useRef<HTMLInputElement>(null);
+
+    const modalWidth = clsx('w-full', size && size);
+
+    const dialogPanelClasses = 'w-full rounded';
+    const headerClasses = 'py-m px-l bg-primary-600 rounded-tl-s rounded-tr-s flex';
+    const mainClasses = clsx(
+        'w-full justify-between bg-white px-l pb-l',
+        ContentVariant.SETTINGS === 'settings' && 'w-full flex-row gap-l',
+        ContentVariant.FILE_UPLOAD === 'file_upload' && 'flex flex-col gap-s pb-s pt-l',
```

Comment:
Does this work? Wouldn't it be easier to check on the contentVariant variable instead on the types.. for example:
`{
'settings': 'w-full flex-row gap-l',
'file_upload': 'flex flex-col gap-s pb-s pt-l'
}[contentVariant]`

Also, make the file_upload string camelCase